mlflow/utils/databricks_utils.py

```diff
@@ -738,54 +738,107 @@ def _init_databricks_cli_config_provider(entry_point):
-            api_token = None
-            try:
-                api_token = entry_point.getNonUcApiToken()
-            except Exception:
-                # Using apiToken from command context would return back the token which is not
-                # refreshed.
-                fallback_api_token_option = notebook_utils.getContext().apiToken()
-                logger.logUsage(
-                    "refreshableTokenNotFound",
-                    {"api_url": api_url},
-                    None,
-                )
-                if fallback_api_token_option.isDefined():
-                    api_token = fallback_api_token_option.get()
+                # Invoking getContext() will attempt to find the credentials related to the
+                # current command execution, so it's critical that we execute it on every
+                # get_config().
+                api_url_option = notebook_utils.getContext().apiUrl()
+                api_url = api_url_option.get() if api_url_option.isDefined() else None
+                # Invoking getNonUcApiToken() will attempt to find the current credentials related
+                # to the current command execution and refresh it if its expired automatically,
+                # so it's critical that we execute it on every get_config().
+                api_token = None
+                try:
+                    api_token = entry_point.getNonUcApiToken()
+                except Exception:
+                    # Using apiToken from command context would return back the token which is not
+                    # refreshed.
+                    fallback_api_token_option = notebook_utils.getContext().apiToken()
+                    logger.logUsage(
+                        "refreshableTokenNotFound",
+                        {"api_url": api_url},
+                        None,
+                    )
+                    if fallback_api_token_option.isDefined():
+                        api_token = fallback_api_token_option.get()
 
-            ssl_trust_all = entry_point.getDriverConf().workflowSslTrustAll()
+                ssl_trust_all = entry_point.getDriverConf().workflowSslTrustAll()
 
-            # Fallback to the default behavior if we were unable to find a token.
-            if api_token is None or api_url is None:
-                return DefaultConfigProvider().get_config()
+                # Fallback to the default behavior if we were unable to find a token.
+                if api_token is None or api_url is None:
+                    return DefaultConfigProvider().get_config()
 
-            return DatabricksConfig.from_token(
-                host=api_url, token=api_token, insecure=ssl_trust_all
-            )
+                return DatabricksConfig.from_token(
+                    host=api_url, token=api_token, insecure=ssl_trust_all
+                )
+    elif dbr_major_minor_version >= (9, 0):
+        class DynamicConfigProvider(DatabricksConfigProvider):
```

Comment:
Ref: `spark/driver/python/python_shell/dbruntime/databricksCli.py` in universe 13.1